agent/connect/ca/provider_vault.go

```diff
@@ -285,7 +285,8 @@ func (v *VaultProvider) GenerateRoot() (RootResult, error) {
 	rootPEM, err := v.getCA(v.config.RootPKIPath)
 	switch err {
 	case ErrBackendNotMounted:
-		err := v.client.Sys().Mount(v.config.RootPKIPath, &vaultapi.MountInput{
+
+		err := v.mountNamespaced(v.config.RootPKINamespace, v.config.RootPKIPath, &vaultapi.MountInput{
```

Comment:
We may want to add a comment in here somewhere (maybe in `mountNamespaced` or `unmountNamespaced`?) about how `RootPKINamespace` and `IntermediatePKINamespace` interact with `Namespace`.

I'm a bit confused myself.

If `Namespace` was set to `foo`, and `RootPKINamespace` was set to `foo`, wouldn't the API call actually go to something in the non-existent namespace `foo/foo`? My understanding is that `Namespace` is setting the `X-Vault-Namespace` header, and then any namespace "in the path" ([like set in makePathHelper](https://github.com/hashicorp/consul/pull/12904/files#diff-7bf30611a760296e2e6ffcd7fe955b0eb8317fe0a43c470b8b52d311af4a2515R743)) will be treated as relative to the namespace set in `X-Vault-Namespace`.

If we aren't sure, we probably want to test this case and make sure we understand the behavior... I'm not sure if `RootPKINamespace` and `IntermediatePKINamespace` currently "overrode" `Namespace` as we intended.

Reply:
Good catch as it turns out. This was marked as a draft because I hadn't tested that case, and there were a few fixes needed to get the behavior sorted out. Tested and seems ok now.